docs/source/trainer/using_the_trainer.rst

```diff
@@ -0,0 +1,484 @@
+-  ``def loss(batch, outputs)``: returns the loss based on the
+   ``outputs`` from the forward pass, and the dataloader.
+
+For more information, see the :doc:`ComposerModel</composer_model>` guide.
+
+A minimal example of a ResNet-18 model is shown here:
+
+.. testcode::
+
+   import torchvision
+   import torch.nn.functional as F
+
+   from composer.models import ComposerModel
+
+   class ResNet18(ComposerModel):
+
+       def __init__(self):
+           super().__init__()
+           self.model = torchvision.models.resnet18()
+
+       def forward(self, batch):
+           inputs, _ = batch
+           return self.model(inputs)
+
+       def loss(self, outputs, batch):
+           _, targets = batch
+           return F.cross_entropy(outputs, targets)
+
+Then, the model can be passed to the trainer with the relevant torch
+objects.
+
+.. code:: python
+
+   trainer = Trainer(model=ResNet18(),
+                     train_dataloader=train_dataloader,
+                     eval_dataloader=eval_dataloader,
+                                 optimizers=torch.optim.Adam(lr=0.01),
+                     max_duration=10,  # epochs
+                     device='gpu')
+
+   trainer.fit()
+
+In the background, we automatically add the :class:`.TQDMLogger` to log
+training progress to the console.
+
+
+A few tips and tricks for using our Trainer:
+
+-  For time-related inputs, such as the ``max_duration`` above, we
+   support both an integer (which we assume is epochs), or as a string.
+   For example, ``"10ba"`` means 10 minibatches or steps, and ``"10ep"``
+   denotes 10 epochs.
+-  If you are using gradient accumulation, the ``batch_size`` in your
+   dataloaders should be the macrobatch size — the batch size of your
+   optimization update. For example, with ``grad_accum=2`` and
+   ``batch_size=2048`` , the train runs through two microbatches of 1024
+   each, then performs a gradient update step.
+-  At any time, most of the relevant quantities for debugging are
+   centralized into one variable: :class:`.State`.
+-  We have an easy abstraction of tracking :class:`.Time`, see the Time guide.
+
+For a full list of Trainer options, see :class:`.Trainer`. Below, we
+illustrate some example use cases.
+
+Training Loop
+~~~~~~~~~~~~~
+
+Behind the scenes, our trainer handles much of the engineering for
+distributed training, gradient accumulation, device movement, gradient
+scaling, and others. The *pseudocode* for our trainer loop as it
+interacts with the :class:`.ComposerModel` is as follows:
+
+.. code:: python
+
+   # training loop
+   for batch in train_dataloader:
+
+       outputs = model.forward(batch)
+       loss = model.loss(outputs, batch)
+
+       loss.backward()
+       optimizer.step()
+
+   # eval loop
+   for batch in eval_dataloader:
+       outputs, targets = model.validate(batch)
+       metrics.update(outputs, target)
+
+For the actual code, see :meth:`.Trainer._train_batch_inner` and the :meth:`.Trainer.eval` methods.
```

Comment:
Private methods won't autolink since they are not documented.

```suggestion
For the actual code, see :meth:`.Trainer.fit` and the :meth:`.Trainer.eval` methods.
```